tmc/helpers/Config.h

```diff
@@ -10,6 +10,10 @@
 #ifndef TMC_HELPERS_CONFIG_H_
 #define TMC_HELPERS_CONFIG_H_
 
+// CONFIG_RESET is also a zephyr config macro.
+// Undefine it so it doesn't conflict with the ConfigState enum below
+#undef CONFIG_RESET
```

Comment:
Question: Is this really the best way for us to handle this? Where is this header actually included by us, and could we `#undef` it immediately before including this TMC header?

Reply:
We are not using it at the moment. I agree that we should at least try to get rid of this change here.

Reply:
Will do. It had been necessary when I was using the SPI functions from this API, but it is not anymore